### PR TITLE
feat(EMS-3306-3308): No PDF - Change your answers - Your Buyer - Trading history

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR adds the ability to change the `TRADED_WITH_BUYER` answer from "yes" to "no" and vice versa, via the "Application - Check your answers" page.

This PR also introduces various E2E improvements from changing the "Your buyer" answers.

## Resolution :heavy_check_mark:
- Fix/update some expected yes/no values in`checkYourBusinessSummaryList`.
- Create new `assertEmptyTradingHistoryFieldValues` cypress command.
- Add E2E test coverage in "Application - Check/change your answers - Your buyer" for:
  - Changing currencies.
  - Changing `HAS_BUYER_FINANCIAL_ACCOUNTS`.
  - Changing credit insurance history answers.
  - Changing `FAILED_PAYMENTS`
  - Changing `OUTSTANDING_PAYMENTS` from "no" to "yes".
  - Changing `OUTSTANDING_PAYMENTS` from "yes" to "no" and then back to "yes".
  - Changing `TRADED_WITH_BUYER` from "no" to "yes".
  - Changing `TRADED_WITH_BUYER` from "yes" to "no" and then back to "yes".
- Split up E2E tests in "Application - Your buyer - Check/change your answers".
- All "Change your answers" E2E tests for Your buyer are now aligned and split up.
- Update the `TRADED_WITH_BUYER` POST controller to redirect appropriately if the URL is a "check and change" route and an answer of "yes" has been submitted.

## Miscellaneous :heavy_plus_sign:
- Fix some invalid paths in `test.yml`.
- Replace all instances of "Your Buyer" with "Your buyer".
- Align "Your buyer - change your answers" E2E test descriptions and file names.
- Add some missing task status assertions in "Application - Check/change your answers - Your buyer" E2E tests.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -298,7 +298,7 @@ jobs:
             "submit-textarea-fields-with-special-characters/**/*.spec.js",
             "task-list/**/*.spec.js",
             "your-business/alternative-trading-address/**/*.spec.js",
-            "your-business/change-your-answers/**/*.spec.js",
```
Comment (Zainzzkk) [inline .github/workflows/test.yml:301]: they should still all run with this?

Comment (ttbarnes) [reply .github/workflows/test.yml:301]: Whoops you're right, I thought this was failing in the main branch but it's not. Reverted 👍

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -307,7 +307,7 @@ jobs:
             "your-business/turnover/**/*.spec.js",
             "your-buyer/alternative-currency/**/*.spec.js",
             "your-buyer/buyer-financial-information/**/*.spec.js",
-            "your-buyer/change-your-answers/**/*.spec.js",
+            "your-buyer/change-your-answers/*.spec.js",
```
Comment (Zainzzkk) [inline .github/workflows/test.yml:310]: this can be changed too :P

**Review by Zainzzkk — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (ttbarnes) [reply .github/workflows/test.yml:310]: Will do in the next PR 👍

**Pull request merged**